js/kubewarden/host_capabilities/oci/manifest_digest/manifest_digest.test.ts

```diff
@@ -0,0 +1,104 @@
+import { ManifestDigest } from "./manifest_digest";
+import * as HostCall from "../../index";
+
+class MockWasmHost {
+    private knownImages: Map<string, string> = new Map();
+
+    constructor(imageDigests: Record<string, string> = {}) {
+        Object.entries(imageDigests).forEach(([image, digest]) => {
+            this.knownImages.set(image, digest);
+        });
+    }
+
+    hostCall(
+        binding: string,
+        namespace: string,
+        operation: string,
+        payload: ArrayBuffer
+    ): Uint8Array {
+        if (
+            binding !== "kubewarden" ||
+            namespace !== "oci" ||
+            operation !== "v1/manifest_digest"
+        ) {
+            throw new Error("Invalid host call parameters");
+        }
+
+        const image = JSON.parse(new TextDecoder().decode(payload)) as string;
+
+        if (this.knownImages.has(image)) {
+            const digest = this.knownImages.get(image)!;
+            return new TextEncoder().encode(JSON.stringify({ digest }));
+        }
+
+        throw new Error(`Image not found: ${image}`);
+    }
+}
+
+describe("ManifestDigest Unit Tests", () => {
+    let mockHost: MockWasmHost;
+    let originalHostCall: typeof HostCall.HostCall.hostCall;
+
+    beforeEach(() => {
+        originalHostCall = HostCall.HostCall.hostCall;
+    });
+
+    afterEach(() => {
+        HostCall.HostCall.hostCall = originalHostCall;
+        jest.restoreAllMocks();
+    });
+
+    it("should return digest for valid image", () => {
+        const image = "myimage:latest";
+        const expectedDigest = `sha256:${image.replace(/[^a-zA-Z0-9]/g, "")}`;
```

Comment:
I don't understand why you're building the expectedDigest in this way. I would just assign a hard coded value (which doesn't necessary have to be a valid shasum, it could just be `sha256:123456`)